### PR TITLE
Test/add toggleable tag hamburger

### PR DESCRIPTION
**Pull request by molnosikristof**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "develop": "gatsby develop --host localhost --port 8000",
     "format": "prettier --trailing-comma es5 --no-semi --single-quote --write 'src/**/*.js'",
     "test": "npm run lint",
-    "test:e2e": "E2E_OUTPUT=${E2E_OUTPUT-./e2e/test_results} E2E_ERRORSHOTS_OUTPUT=${E2E_ERRORSHOTS_OUTPUT-./e2e/errorShots} E2E_SCREENSHOTS=${E2E_SCREENSHOTS-./e2e/screenshots/} wdio",
+    "test:e2e": "E2E_OUTPUT=${E2E_OUTPUT-./e2e/test_results} E2E_ERRORSHOTS_OUTPUT=${E2E_ERRORSHOTS_OUTPUT-./e2e/errorShots} E2E_SCREENSHOTS=${E2E_SCREENSHOTS-./e2e/screenshots/} E2EPROFILE=${E2EPROFILE-sauceextended} wdio",
```
Comment (oroce) [inline package.json]: this default will be the default on local computers so this should just be, chrome, right?

Comment (molnosikristof) [reply package.json]: If you write nothing on a local computer (where CI != truthy or TEST_PROVIDER != sauce) this runs headless chrome. If you use e.g. E2EPROFILE=firefox the tests will run on Firefox.

**Comment by oroce** [comment]
how can that happen?


`E2EPROFILE=${E2EPROFILE-sauceextended}` evaluates to sauceextended if no
profile is set.

**Comment by molnosikristof** [comment]
```
(process.env.CI || process.env.TEST_PROVIDER === 'sauce') ? sauceBrowsers : localBrowsers,
```

If you don't run on sauce you will run the localBrowsers which chrome by default.

File: package.json (hunk 1 above)
Comment (oroce) [reply package.json]: didn't we stop using `TEST_PROVIDER`?

Comment (oroce) [reply package.json]: didn't we agree that we remove the `CI` check?

**Comment by molnosikristof** [comment]
We have only E2EPEOFILE now. :)

File: wdio.conf.js [hunk 2]
```diff
@@ -9,32 +9,78 @@ const { getScreenshotName, getRefPicName, getDiffScreenshotName } = visualRegres
...
 let sauceLabsUsername;
 let saucelabsAccesKey;
 let driver = 'selenium-standalone';
-let localBrowsers = [];
-let sauceBrowsers;
+let browsers = [];
 
-if (process.env.CI || process.env.TEST_PROVIDER === 'sauce') {
+if (e2eProfile.includes('sauce')) {
+  console.log('sauce');
```
Comment (oroce) [inline wdio.conf.js]: please remove the console.log

File: wdio.conf.js [hunk 3]
```diff
@@ -9,32 +9,78 @@ const { getScreenshotName, getRefPicName, getDiffScreenshotName } = visualRegres
 const resolution = { width: 1400, height: 1050 };
 const screenResolution = resolution.width.toString() + 'x' + resolution.height.toString();
 const browserName = 'chrome';
+const e2eProfile = (process.env.E2EPROFILE) ? process.env.E2EPROFILE : '';
```
Comment (oroce) [inline wdio.conf.js]: javascript is for lazy people: `process.env.E2EPROFILE || ''` works as well:)

File: wdio.conf.js [hunk 4]
```diff
@@ -9,32 +9,78 @@ const { getScreenshotName, getRefPicName, getDiffScreenshotName } = visualRegres
...
+      platform: 'Windows 10'
+    }, {
+      browserName: 'chrome',
+      version: 'latest-1',
+      screenResolution: screenResolution,
+      platform: 'macOS 10.13'
+    }];
+  }
+} else if (!e2eProfile.includes('sauce') && e2eProfile !== '') {
+  console.log('not sauce');
```
Comment (oroce) [inline wdio.conf.js]: please remove this

File: wdio.conf.js [hunk 5]
```diff
@@ -9,32 +9,78 @@ const { getScreenshotName, getRefPicName, getDiffScreenshotName } = visualRegres
...
+  console.log('sauce');
   isDefaultBrowser = false;
   sauceLabsUsername = process.env.SAUCE_LABS_USERNAME;
   saucelabsAccesKey = process.env.SAUCE_LABS_ACCESS_KEY;
   driver = 'sauce';
-}
 
-if (process.env.E2EPROFILE && process.env.E2EPROFILE !== 'saucelight'
-    && process.env.E2EPROFILE !== 'sauceextended') {
+  if (e2eProfile === 'saucelight') {
```
Comment (oroce) [inline wdio.conf.js]: to be honest this seems for me a bit easier to understand:

```
const e2eProfile = (process.env.E2EPROFILE || '').split(',');
let isDefaultBrowser = true;
const services = [];
const browsers = [];

if (e2eProfile.includes('saucelight')) {
  services.push('sauce');
  browsers.push({
    browserName: 'chrome',
    version: 'latest',
    screenResolution: screenResolution,
    platform: 'macOS 10.13'
  });
}

if (e2eProfile.includes('sauceextended')) {
  services.push('sauce');
  browsers.push.apply(browsers, [{
    browserName: 'firefox',
    version: 'latest',
    screenResolution: screenResolution,
    platform: 'macOS 10.13'
  }, {
    browserName: 'chrome',
    version: 'latest',
    screenResolution: screenResolution,
    platform: 'macOS 10.13'
  }, {
    browserName: 'chrome',
    'chromeOptions': {
      'args': ['disable-infobars']
    },
    version: 'latest-1',
    screenResolution: screenResolution,
    platform: 'Windows 10'
  }, {
    browserName: 'firefox',
    version: 'latest',
    screenResolution: screenResolution,
    platform: 'Windows 10'
  }, {
    browserName: 'chrome',
    'chromeOptions': {
      'args': ['disable-infobars']
    },
    version: 'latest',
    screenResolution: screenResolution,
    platform: 'Windows 10'
  }, {
    browserName: 'chrome',
    version: 'latest-1',
    screenResolution: screenResolution,
    platform: 'macOS 10.13'
  }]);
}

if (e2eProfile.includes('chrome')) {
  services.push('selenium-standalone');
  browsers.push({
    width: resolution.width,
    height: resolution.height,
    browserName: browserName,
    chromeOptions: {
      'args': ['disable-infobars']
    }
  });
}

if (e2eProfile.includes('chromeheadless')) {
  services.push('selenium-standalone');
  browsers.push({
    width: resolution.width,
    height: resolution.height,
    browserName: 'chrome',
    chromeOptions: {
      'args': ['disable-infobars', '--headless']
    }
  });
}

if (e2eProfile.includes('firefox')) {
  services.push('selenium-standalone');
  browsers.push({
    width: resolution.width,
    height: resolution.height,
    browserName: 'firefox',
    chromeOptions: {
      'args': ['disable-infobars']
    }
  });
}

if (e2eProfile.includes('firefoxheadless')) {
  services.push('selenium-standalone');
  browsers.push({
    width: resolution.width,
    height: resolution.height,
    browserName: 'firefox',
    'moz:firefoxOptions': {
      // flag to activate Firefox headless mode (see https://github.com/mozilla/geckodriver/blob/master/README.md#firefox-capabilities for more details about moz:firefoxOptions)
      args: ['-headless']
    }
  });
}

const isDefaultBrowser = drivers.includes('sauce') === false;
module.exports = {
  services: services.filter((service, ndx) => ndx === services.indexOf(service)), // array.unique
  user: process.env.SAUCE_USERNAME,
  key: process.env.SAUCE_ACCESS_KEY,
}
```

File: wdio.conf.js [hunk 6]
```diff
@@ -8,94 +8,121 @@ const { getScreenshotName, getRefPicName, getDiffScreenshotName } = visualRegres
 
 const resolution = { width: 1400, height: 1050 };
 const screenResolution = resolution.width.toString() + 'x' + resolution.height.toString();
-const browserName = 'chrome';
-const e2eProfile = process.env.E2EPROFILE || '';
+const e2eProfile = (process.env.E2EPROFILE || '').split(',');
```
Comment (oroce) [inline wdio.conf.js]: `const e2eProfile = (process.env.E2EPROFILE || '').split(',').filter(profile => profile !== '')`

so later you don't need to:
`if (e2eProfile[0] !== '')`

you can just do:

`if (e2eProfiles.length === 0) { // default `

File: wdio.conf.js [hunk 7]
```diff
@@ -8,94 +8,121 @@ const { getScreenshotName, getRefPicName, getDiffScreenshotName } = visualRegres
...
-      });
-    } else {
-      browsers.push({
-        width: resolution.width,
-        height: resolution.height,
-        browserName: element,
-        chromeOptions: {
-          'args': ['disable-infobars']
-        } });
+  sauceLabsUsername = process.env.SAUCE_LABS_USERNAME;
```
Comment (oroce) [inline wdio.conf.js]: you can move this directly to config i guess, selenium standalone wont be hurt

File: wdio.conf.js [hunk 8]
```diff
@@ -8,94 +8,121 @@ const { getScreenshotName, getRefPicName, getDiffScreenshotName } = visualRegres
...
+    platform: 'Windows 10'
+  }, {
+    browserName: 'chrome',
+    version: 'latest-1',
+    screenResolution: screenResolution,
+    platform: 'macOS 10.13'
+  });
+
+  driver = 'sauce';
   isDefaultBrowser = false;
```
Comment (oroce) [inline wdio.conf.js]: you can check what is the driver and decide based on that whether it's the default browser or not

File: wdio.conf.js [hunk 9]
```diff
@@ -134,7 +154,7 @@ exports.config = {
       misMatchTolerance: 3.0
     }),
   },
-  user: sauceLabsUsername,
+  user: process.env.SAUCE_LABS_USERNAME,
```
Comment (oroce) [inline wdio.conf.js]: can't we include here the `SAUCE_ACCESS_KEY` as well?

**Pull request merged**